src/Symfony/Component/Form/Extension/Core/Type/TimezoneType.php

```diff
@@ -38,14 +38,16 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setDefaults(array(
             'choice_loader' => function (Options $options) {
                 $regions = $options['regions'];
+                $country = $options['country'];
 
-                return new CallbackChoiceLoader(function () use ($regions) {
-                    return self::getTimezones($regions);
+                return new CallbackChoiceLoader(function () use ($regions, $country) {
+                    return self::getTimezones($regions, $country);
                 });
             },
             'choice_translation_domain' => false,
             'input' => 'string',
             'regions' => \DateTimeZone::ALL,
```

Comment:
i think to get the "right" DX we should determine a default (ALL | PER_COUNTRY) based on `country` is set yes/no.